google-map.html

```diff
@@ -378,9 +378,23 @@
     },
 
     _updateMarkers: function() {
-      this.markers = Array.prototype.slice.call(
+      var newMarkers = Array.prototype.slice.call(
           Polymer.dom(this.$.markers).getDistributedNodes());
 
+      // do not recompute if markers have not been added or removed
+      if (newMarkers.length === this.markers.length) {
+        var added = newMarkers.filter( function(m) {
+            return this.markers && this.markers.indexOf(m) === -1;
```

Comment:
That's true for additions as well. Im confused on the filtering. If the length changed, you always want to recompute:

    if (newMarkers.length === this.markers.length) {
      return;
    }